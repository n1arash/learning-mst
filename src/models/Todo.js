import { types } from "mobx-state-tree";
import { v4 } from "uuid";

export const TodoListItem = types
  .model({
    id: types.optional(types.string, v4()),
    title: types.string,
    completed: types.optional(types.boolean, false)
  })
  .actions(self => ({
    changeTitle(title) {
      self.title = title;
    },
    toggleTodo() {
      self.completed = !self.completed;
    }
  }));

export const TodoList = types
  .model({
    items: types.array(TodoListItem)
  })
  .actions(self => ({
    add(item) {
      self.items.push(item);
    }
  }))
  .views(self => ({
    get countAllTodos() {
      return self.items.length;
    }
  }));
