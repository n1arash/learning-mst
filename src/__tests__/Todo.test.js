import { TodoListItem, TodoList } from "../models/Todo";
import { getSnapshot } from "mobx-state-tree";

it("it should create instance of Todo", () => {
  const item = TodoListItem.create({
    title: "MST Todo MVC",
    completed: true
  });
  expect(item.title).toBe("MST Todo MVC");
  item.changeTitle("Redux");
  expect(item.title).toBe("Redux");
  item.toggleTodo();
  expect(item.completed).toBeFalsy();
});

it("Should create List of TodoListItem", () => {
  const list = TodoList.create({
    items: [
      {
        title: "Learn Django"
      }
    ]
  });

  expect(list.items.length).toBe(1);
  expect(list.items[0].completed).toBeFalsy();
});

it("Can Add TodoListItem to TodoList", () => {
  const list = TodoList.create();
  list.add({
    id: "1",
    title: "Start Learning MST",
    completed: true
  });
  expect(getSnapshot(list)).toMatchSnapshot();
});

it("can return count of all TodoListItem in TodoList", () => {
  const list = TodoList.create();
  list.add({
    title: "Hello MST"
  });

  expect(list.countAllTodos).toBe(1);
});
