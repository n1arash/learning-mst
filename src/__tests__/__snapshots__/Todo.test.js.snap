// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can Add TodoListItem to TodoList 1`] = `
Object {
  "items": Array [
    Object {
      "completed": true,
      "id": "1",
      "title": "Start Learning MST",
    },
  ],
}
`;
